### PR TITLE
Modernize and minimize pylint configuration

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

This updates the pylint configuration. The new configuration only contains settings that we have changed from the default along with a description of why.

This pull request should not contain any changes to the actual linter settings. Using a [minimal configuration](https://pylint.pycqa.org/en/latest/user_guide/configuration/index.html#configuration) is recommended in the pylint documentation. I suspect this is because it is more clear and avoids problems when pylint is updated and older settings are removed (our current configuration generates a stream of warnings about settings that are no longer available).

@ESMValGroup/technical-lead-development-team 

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [ ] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by valeriupredoi** [comment]
@bouweandela what's the status on this? It look OK to me apart from a missing label or two, and a mergy-merge of `main` - if you want this in 2.7 could you pls have a look, do the stuff, and let me know if you OK to get it in 2.7 :beer:

**Comment by bouweandela** [comment]
I added a label and merged main. I think this is ready for review.

**Comment by valeriupredoi** [comment]
thanky much @bouweandela - am just rerunning a test that failed for some obscure internal reasons, then I'll go have me a review if that's ok with you :beer:

**Review by valeriupredoi — APPROVED** [review]
looks good to me, @zklaus would you be able to have a look too, pls mate?

**Comment by valeriupredoi** [comment]
small ping @zklaus - would you have time to look at this soon, bud? Am releasing rc1 today but we can prob sneak it in rc2 or stable if you have a chance to look at it :beer:

File: pyproject.toml [hunk 1]
```diff
@@ -3,3 +3,23 @@ requires = ["setuptools >= 40.6.0", "wheel", "setuptools_scm>=6.2"]
...
+    "logger",       # Our preferred name for the logger
+]
+[tool.pylint.format]
+max-line-length = 79
+[tool.pylint."messages control"]
+disable = [
+    "import-error",      # Needed because Codacy does not install dependencies
+    "file-ignored",      # Disable messages about disabling checks
+    "locally-disabled",  # Disable messages about disabling checks
+]
```
[the remark this answers is not in the document]
Comment (bouweandela) [reply pyproject.toml:25]: Maybe, do you know what the name is that we would need to use here to disable the Django error message?

**Review by zklaus — APPROVED** [review]
I think it's ok. I like the general approach. Initially I wanted to suggest improvements to the comments on message control, but didn't come up with anything better.

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (valeriupredoi) [reply pyproject.toml:25]: not sure, let's just let it through for now, figure it later

**Comment by valeriupredoi** [comment]
she's going in, fellas :beer:

**Pull request merged**